Lookup: Add a Getting Started tutorial (#3289)
DevExpress/devextreme-documentation: concepts/05 UI Components/Lookup/00 Getting Started with Lookup/25 Group Data.md

You can group data by category in the Lookup. To implement this, we will use the data from the previous steps with the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component that allows you to sort, filter, select, and group data.

> To group data, specify the data field to group by the **DataSource**'s [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property and set the Lookup's [grouped](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#grouped) property to **true**.

Reviewer: ```suggestion
Use the **DataSource**'s [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to specify the data field to group by and enable Lookup's [grouped](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#grouped) property to notify the component that the data source is grouped.
```

Author: Noted

> If the number of items doesn't fit in the Lookup size, the scrolling bar appears. You can load items by parts on the "next" button click. Follow the steps below to enable this functionality:

Reviewer: Paging is a different feature, it's not part of grouping.

Author: Noted

    <!-- tab: app.component.ts -->
    // ...
    export class AppComponent {
        // ...
        constructor(service: Service) {
            // ...
            this.dataSource = new DataSource({
                // ...
                group: "Assigned",
                paginate: true,
                pageSize: 2
            });
        }
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <Dx{WidgetName} ...
            :grouped="true"
            page-load-mode="nextButton"
            next-button-text="More"
        />
    </template>

    <script>
        // ...
        const dataSource = new DataSource({
            // ...
            group: "Assigned",
            paginate: true,
            pageSize: 2
        });
    </script>

##### React

    <!-- tab: App.js -->
    // ...
    const dataSource = new DataSource({
        // ...
        group: "Assigned",
        paginate: true,
        pageSize: 2
    });
    
    function App() {
        return (
            <{WidgetName} ...
                grouped="true"
                pageLoadMode="nextButton"
                nextButtonText="More"
            />
        );
    }
    export default App;

> ---

Reviewer: I think we should move info about groupTemplate here, leaving only itemTemplate in the article on customization.

Author: Noted

> You can group data by category in Lookup. To implement this, we will use data from the previous steps with the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component that allows you to sort, filter, select, and group data.

Reviewer 2: ```suggestion
You can group data by category in Lookup. To implement this, we use data from the previous steps with the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component that allows you to sort, filter, select, and group data.
```

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        $("#lookup").dxLookup({
            dataSource: new DevExpress.data.DataSource({ 
                // ...
                group: "Assignee"
            }),
            // ...
            grouped: true
        });
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-lookup ...
        [grouped]="true">
    </dx-lookup>

    <!-- tab: app.component.ts -->
    // ...
    export class AppComponent {
        // ...
        constructor(service: Service) {
            // ...
            this.dataSource = new DataSource({
                // ...
                group: "Assignee"
            });
        }
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxLookup ...
            :grouped="true"
        />
    </template>

    <script>
        // ...
        const dataSource = new DataSource({
            // ...
            group: "Assignee"
        });
    </script>

##### React

    <!-- tab: App.js -->
    // ...
    const dataSource = new DataSource({
        // ...
        group: "Assignee"
    });
    
    function App() {
        return (
            <Lookup ...

>                 grouped="true"

Reviewer: ```suggestion
                grouped={true}
```

            />
        );
    }
    export default App;

---

> To customize group titles appearance, use [groupTemplate](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#groupTemplate).

Reviewer: ```suggestion
To customize the appearance of group titles, use [groupTemplate](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#groupTemplate).
```

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        $("#lookup").dxLookup({
            // ... 
            groupTemplate: function (data, index, element) {
                return data.key + " - " + data.items.length;
            }
        });
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-lookup>

>         // ...

Reviewer: ```suggestion
        <!-- ... -->
```
